index.js

```diff
@@ -79,12 +84,16 @@ Hooks.on("deleteToken", () => {
 Hooks.on("updateScene", (scene, data) => {
     // Skip if the updated scene isn't the current one
     if (scene.id !== canvas.scene.id) return;
-    
+
     if (data.flags && "world-explorer" in data.flags) {
         const worldExplorerFlags = data.flags["world-explorer"];
 
-        // If the only change was revealed positions, do the throttled refresh to not interfere with token moving
-        if (worldExplorerFlags.revealedPositions && Object.keys(worldExplorerFlags).length === 1) {
+        // If the change only affects the mask, do the throttled refresh to not interfere with token moving
+        const maskOnlyFlags = ["gridData", "opacityGM", "opacityPlayer", "partialOpacityGM", "partialOpacityPlayer"];
```

Comment:
What key are you trying to exclude by checking mask only flags?

Reply:
I'm not sure I understand the question.
What I tried to do here is to have stuff that only affects the mask not trigger a full re-draw of the layer, hopefully improving on performance.

This way it only refreshes the mask when the `reveal` of a tile changes or only the opacity setting changes (i.e. through the GM opacity slider). Any other change, like the overlay colour, will still trigger a full redraw.
Essentially, any change made in the scene settings will trigger a full redraw, while changes through the World Explorer canvas tools only refreshes the mask.